src/Bundle/DependencyInjection/Compiler/RegisterFiltersPass.php

```diff
@@ -46,18 +59,29 @@ public function process(ContainerBuilder $container): void
                 $formType = $class::getFormType();
             }
 
-            foreach ($attributes as $attribute) {
-                if (null === $type && null === ($attribute['type'] ?? null)) {
-                    throw new InvalidArgumentException(sprintf('Tagged grid filters needs to have "type" attributes or implements "%s".', TypeAwareFilterInterface::class));
-                }
+            $this->registerFilter($filterRegistry, $formTypeRegistry, $id, $attributes, $type, $formType);
+        }
+    }
 
-                if (null === $formType && null === ($attribute['form_type'] ?? null)) {
-                    throw new InvalidArgumentException(sprintf('Tagged grid filters needs to have "form_type" attributes or implements %s.', FormTypeAwareFilterInterface::class));
-                }
+    private function registerFilter(
+        Definition $filterRegistry,
+        Definition $formTypeRegistry,
+        string $id,
+        array $attributes,
+        ?string $type = null,
+        ?string $formType = null,
+    ): void {
+        foreach ($attributes as $attribute) {
+            if (null === $type && null === ($attribute['type'] ?? null)) {
+                throw new InvalidArgumentException(sprintf('Tagged grid filters needs to have "type" attribute or implements "%s".', TypeAwareFilterInterface::class));
+            }
 
-                $registry->addMethodCall('register', [$type ?? $attribute['type'], new Reference($id)]);
-                $formTypeRegistry->addMethodCall('add', [$type ?? $attribute['type'], 'default', $formType ?? $attribute['form_type']]);
+            if (null === $formType && null === ($attribute['form_type'] ?? null)) {
+                throw new InvalidArgumentException(sprintf('Tagged grid filters needs to have "form_type" attribute or implements "%s".', FormTypeAwareFilterInterface::class));
```

Comment:
According to SF CS, which is basis for sylius CS, global classes should be used directly and not added to `use`.

```suggestion
                throw new \InvalidArgumentException(sprintf('Tagged grid filters needs to have "form_type" attribute or implements "%s".', FormTypeAwareFilterInterface::class));
```

Reply:
Shouldn't this be enforced by our CS rules ?